### PR TITLE
[MINT-2128] Update status modal

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# MINT-2128

## Description

[Figma](https://www.figma.com/design/qXhO12xSSkDF4LaToGKJoV/MINT-Blue-Sky-Ideas?node-id=690-24765&t=LZkX0Ka81xu68h2E-0)

- Added a modal to the task list (Collab area when built) if the presence of an `suggestedPhase`
- Modal will also only render if the user has not viewed this modal in the current session
- Modal will render statuses choose from based on the phase - should match [Figma](https://www.figma.com/design/qXhO12xSSkDF4LaToGKJoV/MINT-Blue-Sky-Ideas?node-id=691-25495&t=59uOkSge6oRs6Y8S-0)
- Unit tests for session and modal

## How to test this change

1.  Seed db
2. Click on `Enhancing Oncology Model`, verify that the modal renders and does not render again once the modal is closed.  The modal should render again on re-login if the status was not updated
3. Test the other phases based on the criteria set in Figma

## PR Author Checklist

- [x] I have provided a detailed description of the changes in this PR.
- [x] I have provided clear instructions on how to test the changes in this PR.
- [x] I have updated tests or written new tests as appropriate in this PR.
- [x] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: src/components/UpdateStatusModal/__snapshots__/index.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
+
+exports[`UpdateStatusModal > matches snapshot 1`] = `<DocumentFragment />`;
```
Comment (garyjzhao) [inline src/components/UpdateStatusModal/__snapshots__/index.test.tsx.snap:3]: This doesn't look right. Snapshot is not capturing the right things

File: src/components/UpdateStatusModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            type="button"
+            className="margin-left-2 text-red"
+            data-testid="go-to-timeline"
+            unstyled
+            onClick={() => {
+              closeModal();
+            }}
+          >
+            {modelPlanTaskListT('statusModal.goToTimeline')}
+          </Button>
```
Comment (garyjzhao) [inline src/components/UpdateStatusModal/index.tsx:170]: Figma has the link in blue and, according to Zoe's comment, go to the timeline 

image

**Review by garyjzhao — CHANGES_REQUESTED** [review]
Overall looks good and functions well! Just a few things

File: src/components/UpdateStatusModal/index.tsx (hunk 2 above)
Comment (patrickseguraoddball) [reply src/components/UpdateStatusModal/index.tsx:170]: Ahh I was reading it as task list lol, nice catch

**Review by garyjzhao — APPROVED** [review]
Thanks for the changes. LGreatTM

**Pull request merged**